test/News.test.js

```diff
@@ -0,0 +1,223 @@
+let chai = require('chai');
+let chaiHttp = require('chai-http');
+const expect = require('chai').expect;
+const { Entry, Category } = require('../database/models');
+
+chai.use(chaiHttp);
+const url= 'http://localhost:3000';
+
+let newsTest = {
+    name: "test",
+    image: "urlTest",
+    content: "ContentTest",
+}
+
+
+describe('Get all news: ',()=>{
+    let idNewTest;
+    before(async () => {
+        const newEntry = new Entry(newsTest);
+        await newEntry.save();
+        idNewTest = newEntry.dataValues.id;
+    })
+
+    it('should get all news', (done) => {
+        chai.request(url)
+        .get('/news')
+        .end( (err,res) => {
+            expect(res).to.have.status(200);
+            expect(err).to.be.equal(null);
+        done();
+    });
+    });
+
+    it('should get json news', (done) => {
+        chai.request(url)
+        .get('/news')
+        .end( (err,res) => {
+            expect(res).to.be.json;
+            expect(res.body.body).to.be.a('array')
+        done();
+    });
+    });
+    
+  
+    it('should get a new by Id', (done) => {
+        chai.request(url)
+        .get('/news/' + idNewTest)
+        .send()
+        .end( (err,res) => {
+            expect(res.body.body).to.have.property('id').to.be.equal(idNewTest);
+            expect(res).to.have.status(200);
+        done();
+    });
+    });
+        
+    it('id should be a number', (done) => {
+        chai.request(url)
+        .get('/news/' + idNewTest)
+        .send()
+        .end( (err,res) => {
+            expect(idNewTest).to.be.a('number');
+        done();
+    });
+    });
+    
+    
+    after(async () => {
+        const news = await Entry.destroy({
+            where: { id: idNewTest },
+            force: true
+        });
+    });
+
+});
+   
+
+describe('Post a new: ',()=>{
+   let idTestCategory;
+   let newsTestPost;
+   let idNewTest;
+
+   before(async () => {
+        let categoryTest = {
+            name: "categoryTest",
+            description: "descriptionCategoryTest",
+        }
+    const newCategory = new Category(categoryTest);
+    await newCategory.save();
+    idTestCategory = newCategory.dataValues.id;
+     newsTestPost = {
+        name: "test",
+        image: "urlImageTest",
+        content: "contentTest",
+        categoryId: idTestCategory,
+        }
+    })
+
+
+    it('should post a new', (done) => {
+        chai.request(url)
+        .post('/news')
+        .send(newsTestPost)
+        .end( (err,res) => {
+            idNewTest = res.body.body.id
+            expect(res).to.have.status(200);
+        done();
+    });
+    });
+
+    it('should post a new with an existing categoryId', (done) => {
```

Comment:
acá hay un error con el mensaje del it, ya que **no** debe agregar la novedad **si no** existe la categoría